docs/README.8021q.routing.topo.md

```diff
@@ -0,0 +1,465 @@
+sudo ip link add name veth_ac5 type veth peer name ac5
+sudo ip link add name veth_ac6 type veth peer name ac6
+sudo ip link add name veth_trunk1 type veth peer name trunk1
+sudo ip link add name veth_trunk2 type veth peer name trunk2
+
+```
+
+Create network namespace for end hosts and inject one end of veth pair
+
+```
+sudo ip netns add host-1.0
+sudo ip netns add host-2.0
+sudo ip netns add host-3.0
+sudo ip netns add host-4.0
+sudo ip netns add host-5.0
+sudo ip netns add host-6.0
+sudo ip link set dev veth_ac1 netns host-1.0
+sudo ip link set dev veth_ac2 netns host-2.0
+sudo ip link set dev veth_ac3 netns host-3.0
+sudo ip link set dev veth_ac4 netns host-4.0
+sudo ip link set dev veth_ac5 netns host-5.0
+sudo ip link set dev veth_ac6 netns host-6.0
+```
+
+Configure IP addresses inside the host net namepaces
+
+```
+sudo ip netns exec host-1.0 bash
+ip link set dev veth_ac1 up
+ip addr add 10.0.1.1/24 dev veth_ac1
+ip route add 10.0.2.0/24 via 10.0.1.10
+ip address show
+ip route show
+exit
+sudo ip netns exec host-2.0 bash
+ip link set dev veth_ac2 up
+ip addr add 10.0.2.1/24 dev veth_ac2
+ip route add 10.0.1.0/24 via 10.0.2.20
+ip address show
+ip route show
+exit
+sudo ip netns exec host-3.0 bash
+ip link set dev veth_ac3 up
+ip addr add 10.0.3.1/24 dev veth_ac3
+ip address show
+ip route show
+exit
+sudo ip netns exec host-4.0 bash
+ip link set dev veth_ac4 up
+ip addr add 10.0.1.2/24 dev veth_ac4
+ip route add 10.0.2.0/24 via 10.0.1.10
+ip address show
+ip route show
+exit
+sudo ip netns exec host-5.0 bash
+ip link set dev veth_ac5 up
+ip addr add 10.0.2.2/24 dev veth_ac5
+ip route add 10.0.1.0/24 via 10.0.2.20
+ip address show
+ip route show
+exit
+sudo ip netns exec host-6.0 bash
+ip link set dev veth_ac6 up
+ip addr add 10.0.3.2/24 dev veth_ac6
+ip address show
+ip route show
+exit
+```
+
+Now start the sonic container and pass the veth interfaces to sonic-vpp
+```
+sudo ./start_sonic_vpp.sh start -n SONIC-A -i ac1,ac2,ac3,trunk1
+```
+
+start the second sonic container and pass the veth interfaces to sonic-vpp
+```
+sudo ./start_sonic_vpp.sh start -n SONIC-B -i veth_trunk1,veth_trunk2
+```
+
+start the third sonic container and pass the veth interfaces to sonic-vpp
+```
+sudo ./start_sonic_vpp.sh start -n SONIC-C -i ac4,ac5,ac6,trunk2
+```
+
+Get into SONIC-A container and configure vlan and vlan bridging
+
+SONIC-A:
+```
+docker exec -it SONIC-A /bin/bash
+show interface status
+config interface startup Ethernet0
+config interface startup Ethernet1
+config interface startup Ethernet2
+config interface startup Ethernet3
+
+config vlan add 10
+config vlan member add -u 10 Ethernet0
+config vlan member add  10 Ethernet3
+
+config vlan add 20
+config vlan member add -u 20 Ethernet1
+config vlan member add  20 Ethernet3
+
+config vlan add 30
+config vlan member add -u 30 Ethernet2
+config vlan member add  30 Ethernet3
+
+config interface ip add Vlan10 10.0.1.10/24
+config interface ip add Vlan20 10.0.2.20/24
```

Comment:
Is there use case test with IPv6 addresses?

Reply:
IPv6 inter vlan traffic is not going through because of ND resolution issue. The reason for this VPP is responding with NA with 802.1q tag to ipv6 host on access interface which is wrong. Same case works fine for V4 case. Seems issue with VPP, going to talk to VPP folks and debug it based their input. But SONiC programming is right as IPv4 is working fine. SO i would go ahead commit the code and in parallel work with VPP folks to find if there is any issue raised on this already.

Reply:
okay, Thanks

Reply:
Please open github issue for the IPv6 problem you mentioned.